### ISSUE TITLE
Wrong implementation of from_value called

### ISSUE THREAD

**Issue by asib** [blackbeam/rust-mysql-simple#19]
I have the following code to run a `SELECT` query and map the result to a struct vec:

```rust
#[derive(Debug, PartialEq, Eq)]
struct Terminal {
    did: u64,
    tid: String,
    rn_cd: u32,
    rn_a: u32,
}

pool.prep_exec("SELECT d.id, d_v.tid \
    FROM d_v \
    INNER JOIN d on d_v.o_d_id = d.id \
    WHERE d_v.t_is_i = 1 \
    AND d.active = 1", ())
.map(|result| {
    result.map(|x| x.unwrap()).map(|mut row| {
        Terminal {
            did: from_value::<u64>(row.pop().unwrap()),
            tid: from_value::<String>(row.pop().unwrap()),
            rd_cn: 0,
            rd_a: 0,
        }
    }).collect()
}).unwrap();
```

When I run this, the program panics with `Error retrieving String from value`. If I change the first line of the query to `SELECT CONVERT(d.id, CHAR) as id, d_v.tid \`, then this doesn't happen and the program runs fine.

So my question is, why does it appear that the `String` implementation of `from_value` is being called for the first column, when I'm explicitly calling the `u64` implementation with `from_value::<u64>()`?

**Comment by asib** [comment]
Finally realised I have to pop the columns in reverse order to the order I selected them in, so I changed the struct construction to:

```rust
Terminal {
    tid: from_value(row.pop().unwrap()),
    did: from_value(row.pop().unwrap()),
    rn_cn: 0,
    rn_a: 0,
}
```

**Comment by blackbeam** [comment]
I have plans to provide `FromRow` trait to solve this.

**Comment by blackbeam** [comment]
`FromRow` now available (in version 0.17.0)
